### ISSUE TITLE
support read only connection datasource

### ISSUE THREAD

**Issue by xenoterracide** [spring-projects/spring-boot#8189]
read only slaves are a fairly common thing. It would be nice if spring boot could support this configuration out of the box for transactions marked as read only with perhaps a secondary "array" of read only datasource configurations (if you have more than one slave).

**Comment by wilkinsona** [comment]
Thanks for the suggestion. Unfortunately, I don't think we really want to get into the business of auto-configuring multiple data sources. Figuring out the user's intent would start to get very complicated when something like Hibernate is thrown into the mix as well. We have [an open issue](https://github.com/spring-projects/spring-boot/issues/3456) to provide an example of how to configure multiple data sources. It actually links to a couple of existing examples that may be of interest.

Routing read-only requests to a slave is only one example of dynamic routing and there are a huge number of other possibilities. Given this, I don't think it makes sense for Boot to provide one specific custom routing logic implementation. I think that Spring Framework's already got this right. It provides `AbstractRoutingDataSource` as helpful base class for your custom logic and then leaves you just to write the logic that suits your specific needs.

To summarise, the recommendation is:

 - Configure multiple data sources manually to meet your needs
 - Use AbstractRoutingDataSource as the foundation for your custom routing logic

**Comment by andyfeller** [comment]
@wilkinsona How would you recommend configuring a single data source as readOnly?

I'm autowiring beans via @ConfigurationProperties annotation to instantiate an arbitrary number of DataSourceProperties instances where I occasionally need to mark 1+ of them as readOnly.  I'm not using @Transactional annotation as I need this across the board for a given data source.

Thanks!

**Comment by wilkinsona** [comment]
@andyfeller This isn't a great place for questions. We prefer that people use Gitter or Stack Overflow. That said, it depends on your database, doesn't it? I'd probably configure a Datasource with credentials for a user that only has read permissions.

**Comment by martin-g** [comment]
@wilkinsona Is Gitter a good place ? I am not able to find a way to search in its archives. If it is not possible to use its history then the provided solutions there are not very useful. People will ask the same questions again and again.

**Comment by wilkinsona** [comment]
You can search on Gitter. There's a magnifying glass icon on the left-hand side of its UI.
